### PR TITLE
Adding models and client for tanzukubernetescluster resource implementation in terraform

### PR DESCRIPTION
**Pull request by ramya-bangera**
1. **What this PR does / why we need it**:
         Includes new models and client for tanzukubernetescluster resource implementation
2. **Which issue(s) this PR fixes**

        (optional, in `fixes #<issue number>` format, will close the issue(s) when PR gets merged):

        Fixes #

3. **Additional information**

4. **Special notes for your reviewer**

### CODE REVIEW

File: internal/models/tanzukubernetescluster/common/cluster_osimage.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+package tkccommon
+
+import (
+	"github.com/go-openapi/swag"
+)
+
+// VmwareTanzuManageV1alpha1ManagementclusterProvisionerTanzukubernetesclusterCommonClusterOSImage The OS image configuration.
+//
+// swagger:model vmware.tanzu.manage.v1alpha1.managementcluster.provisioner.tanzukubernetescluster.common.cluster.OSImage
+type VmwareTanzuManageV1alpha1ManagementclusterProvisionerTanzukubernetesclusterCommonClusterOSImage struct {
```
Comment (Vasundharashukla) [inline internal/models/tanzukubernetescluster/common/cluster_osimage.go:15]: I believe this is common and can be reused from other legacy cluster models.

File: internal/models/tanzukubernetescluster/create_cluster_response.go [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+package tkcmodels
+
+import (
+	"github.com/go-openapi/swag"
+)
+
+// VmwareTanzuManageV1alpha1ManagementclusterProvisionerTanzukubernetesclusterCreateTanzuKubernetesClusterResponse Response from creating a TanzuKubernetesCluster.
+//
+// swagger:model vmware.tanzu.manage.v1alpha1.managementcluster.provisioner.tanzukubernetescluster.CreateTanzuKubernetesClusterResponse
+type VmwareTanzuManageV1alpha1ManagementclusterProvisionerTanzukubernetesclusterCreateTanzuKubernetesClusterResponse struct {
```
Comment (Vasundharashukla) [inline internal/models/tanzukubernetescluster/create_cluster_response.go:15]: This can be added to the same file as cluster_request struct

**Review by Vasundharashukla — CHANGES_REQUESTED** [review]

File: internal/models/tanzukubernetescluster/common/cluster_osimage.go (hunk 1 above)
Comment (ramya-bangera) [reply internal/models/tanzukubernetescluster/common/cluster_osimage.go:15]: I couldn't find it in the existing models. Can you point me to that file?

File: internal/models/tanzukubernetescluster/create_cluster_response.go (hunk 2 above)
Comment (ramya-bangera) [reply internal/models/tanzukubernetescluster/create_cluster_response.go]: Yes moved it to a single file method_create.go like other resources

File: internal/models/tanzukubernetescluster/common/cluster_osimage.go (hunk 1 above)
Comment (Vasundharashukla) [reply internal/models/tanzukubernetescluster/common/cluster_osimage.go:15]: Never mind, that file has other fields too. Let it be this way.

File: internal/client/tanzukubernetescluster/tanzukubernetescluster_resource.go [hunk 3]
```diff
@@ -0,0 +1,131 @@
+/*
+Copyright © 2021 VMware, Inc. All Rights Reserved.
```
[the remark this answers is not in the document]
Comment (ramya-bangera) [reply internal/client/tanzukubernetescluster/tanzukubernetescluster_resource.go]: Fixed

File: internal/client/tanzukubernetescluster/tanzukubernetescluster_resource.go [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+import (
+	"net/url"
+
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/client/transport"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/helper"
+	tkcmodels "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/tanzukubernetescluster"
+)
+
+const (
+	apiVersionAndGroup      = "/v1alpha1/managementclusters"
```
Comment (Vasundharashukla) [inline internal/client/tanzukubernetescluster/tanzukubernetescluster_resource.go:17]: The `/` before v1aplpha1 is needed?

Comment (ramya-bangera) [reply internal/client/tanzukubernetescluster/tanzukubernetescluster_resource.go:17]: Not needed, but it should be no harm as it is trimmed while url is formed. https://github.com/vmware/terraform-provider-tanzu-mission-control/blob/main/internal/client/transport/methods.go#L41

**Pull request closed without merging**